### PR TITLE
Fixup ast_transform docs

### PR DESCRIPTION
**Pull request by esc**
So that they render and read well as HTML.

### CODE REVIEW

**Comment by esc** [comment]
Renders as: https://numba-rvsdg--128.org.readthedocs.build/en/128/reference/datastructures.html#module-numba_rvsdg.core.datastructures.ast_transforms

File: numba_rvsdg/core/datastructures/ast_transforms.py [hunk 1]
```diff
@@ -490,73 +490,73 @@ def function(a: int) -> None
...
+         *  ``i = None                          ## assign target, i``
 
         The next three is for the header, the predicate determines the end of
         the loop.
 
-         *      __iter_last_1__ = i           # backup value of i
-         *      i = next(__iterator_1__, '__sentinel__')  # get next i
-         *      if i != '__sentinel__':       # regular iteration
+         *      ``__iter_last_1__ = i           ## backup value of i``
+         *      ``i = next(__iterator_1__, '__sentinel__')  # * get next i``
```
Comment (kc611) [inline numba_rvsdg/core/datastructures/ast_transforms.py:542]: Small nitpick but the rendering of this particular line makes the RHS comment shift a bit more right. Which overall makes it look very haphazard. Could all the RHS comments in this sections be shifted right by the same amount so as to match the indentation of this particular line

**Review by kc611 — CHANGES_REQUESTED** [review]

File: numba_rvsdg/core/datastructures/ast_transforms.py (hunk 1 above)
Comment (esc) [reply numba_rvsdg/core/datastructures/ast_transforms.py:542]: The indentation is such that it is easier to keep the lines in sync with the above as the indentation level is matched. For the output, they render without all the leading whitespace anyway as shown here. I would prefer thus to keep them as is.

Screenshot 2024-05-29 at 14 02 46

**Review by kc611 — APPROVED** [review]
> The indentation is such that it is easier to keep the lines in sync with the above as the indentation level is matched. For the output, they render without all the leading whitespace anyway as shown here. I would prefer thus to keep them as is.

Yeah that sounds fine.

**Comment by esc** [comment]
@kc611 thank you for the review.

**Pull request merged**